ahnlich/ai/src/server/handler.rs

```diff
@@ -574,6 +621,65 @@ impl AiService for AIProxyServer {
         }))
     }
 
+    #[tracing::instrument(skip_all)]
+    async fn convert_store_input_to_embeddings(
+        &self,
+        request: tonic::Request<ConvertStoreInputToEmbeddings>,
+    ) -> Result<tonic::Response<StoreInputToEmbeddingsList>, tonic::Status> {
+        let params = request.into_inner();
+
+        let ai_model = AiModel::try_from(params.model)
+            .map_err(|_| AIProxyError::InputNotSpecified("AI Model Value".to_string()))?;
+
+        let preprocess_action = params
+            .preprocess_action
+            .ok_or_else(|| AIProxyError::InputNotSpecified("Preprocess Action Value".to_string()))
+            .and_then(|val| {
+                PreprocessAction::try_from(val).map_err(|_| {
+                    AIProxyError::InputNotSpecified("Preprocess Action Value".to_string())
+                })
+            })?;
+
+        let index_model_repr: ModelDetails = SupportedModels::from(&ai_model).to_model_details();
+
+        for store_input in &params.store_inputs {
+            let store_input_type: AiStoreInputType = (store_input)
+                .try_into()
+                .map_err(|_| AIProxyError::InputNotSpecified("Store Input Value".to_string()))?;
+
+            if store_input_type != index_model_repr.input_type() {
+                return Err(tonic::Status::failed_precondition(
+                    "Store input type and AI model type don't match",
+                ));
+            }
+        }
+
+        let inputs = Arc::new(params.store_inputs);
+
+        let store_keys = ModelManager::handle_request(
+            &self.model_manager,
+            &ai_model,
+            Arc::clone(&inputs),
+            preprocess_action,
+            InputAction::Index,
+            None,
+        )
+        .await?;
+
```

Comment:
We probably need to ensure first that they are of the same length before we zip... unless we already do that somewhere within `handle_request` but I doubt that